backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -148,6 +166,29 @@ class VolunteerService implements IVolunteerService {
     return userVolunteerDTOs;
   }
 
+  async getCheckInsAndSchedules(
+    volunteerId: string,
+  ): Promise<(CheckInDTO | SchedulingDTO)[]> {
+    const checkIns = await (
+      await this.checkInService.getCheckInsByVolunteerId(volunteerId)
```

Comment:
i think in this same PR you can make another edit in `checkInService.getCheckInsByVolunteerId` function to add a new where clause where `volunteerNeeded` is true. I am saying this cuz I got an error that `"Cannot read property 'split' of null"` cuz VolunteerTime was null and I updated that schedule with that volunteerId through postman. Even though this will not really happen because volunteer cannot be assigned when volunteer is not needed from the frontend, but it will be good to add this. Maybe for edge cases like a donor scheduled with volunteerNeeded, then volunteer assigns themselves that shift, and then donor updates the schedule to not need a volunteer anymore, then I am thinking the way we update our schedules, we wont set volunteerId to null, and then this function will bug out

Reply:
i dont see why the `volunteerId` wouldnt be updated to null if you can at least change `volunteerNeeded` to be false. why is there a volunteerId associated if a volunteer isn't needed? i agree with putting the volunteerNeeded to be true but i think it's a bandaid to a logic that should be changed imo

Reply:
hopefully the new change to account for `null | undefined` and an incorrectly formatted time string is okay

Reply:
yes you are right, when a schedule is changed back to no volunteer, volunteerId should be changed back to `null` which I think (dont quote me on this) we dont do rn. I can create a ticket for that, but its good that we are ensuring it over here as well. Thanks for the changes